research/workflow-engines/airflow.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Airflow
+authors: lbarcziova
+---
+
+- docs: https://airflow.apache.org/docs/apache-airflow/stable/index.html
+
+  > Apache Airflow is a platform used to programmatically author, schedule, and monitor workflows. It allows you to define complex workflows as directed acyclic graphs (DAGs) and execute them on a scheduled basis or triggered by events.
+
+- A DAG is the core concept in Airflow. It's a collection of tasks and their dependencies, organized as a directed acyclic graph. Example:
+
+```python
+from datetime import datetime
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+
+def build_task():
+    pass
+
+def test_task():
+    pass
+
+def dynamic_workflow():
+    dag = DAG(
+        f'dynamic_workflow',
+        start_date=datetime.now(),
+        schedule_interval=None,
+        catchup=False,
+    )
+
+    build = PythonOperator(
+        task_id=f'build',
+        python_callable=build_task,
+        op_args=[],
+        dag=dag,
+    )
+
+    test = PythonOperator(
+        task_id=f'test',
+        python_callable=test_task,
+        op_args=[],
+        dag=dag,
+    )
+
+    # Set task dependencies
+    build >> test
+
+    return dag
+
+dynamic_workflow()
+
+```
+
+- Airflow uses schedulers and executors to manage task execution. The scheduler determines when to execute tasks, while the executor defines how tasks are run. Airflow supports various executors, including SequentialExecutor, LocalExecutor, and CeleryExecutor.
```

Comment:
Do you know how we can make Airflow tasks to be triggered by an external event?

Will this require implementing our tasks in a different way to combine both starting and waiting for the result?